src/bootstrap.php

```diff
@@ -355,6 +358,31 @@ public function add_body_class( $classes ) {
 		return $classes;
 	}
 
+	/**
+	 * Add PDF URLs to Zapier's data(body).
+	 *
+	 * @param array $body
+	 * @param array $feed
+	 * @param array $entry
+	 * @param array $form
+	 *
+	 * @return array
+	 *
+	 * @since 6.1
+	 */
+	public function add_expanded_zapier_support( $body, $feed, $entry, $form ) {
+		$zapier = \GF_Zapier::get_instance();
+		$signer = new Helper\Helper_Url_Signer();
+		$model  = new Model\Model_PDF( $this->gform, $this->log, $this->options, $this->data, $this->misc, $this->notices, $this->templates, $signer );
+
+		/* @todo : Signed urls */
+		foreach ( $form['gfpdf_form_settings'] as $settings ) {
+			$body[ $zapier->get_body_key( $body, $settings['template'] . ' PDF URL' ) ] = $model->get_pdf_url( $settings['id'], $entry['id'] );
```

Comment:
This should be `$settings['label']` instead of `$settings['template']` I think.

Reply:
The fifth property passed to `get_pdf_url()` should be set to `false`

Reply:
To add signed URL support you would duplicate this line, adjust the body key, and then wrap the PDF URL in `$signer->sign( $url, '+1 week' )`.

Repeat this process for both a month and year: `$signer->sign( $url, '+1 month' )` or `$signer->sign( $url, '+1 year' )`.

Reply:
Copy @jakejackson1